File: src/utils/aiEvaluator.ts
import axios from 'axios';
import type { SalesScore } from './salesDataLoader';

interface EvaluationResponse {
    result: string;
    error?: string;
}

export const getAIEvaluation = async (salesData: SalesScore): Promise<string> => {
    try {
        const message = `请整合以下一或多段录音评价理由，列出1234但不要加粗或变为标题格式，保留一些录音中的例子，不要出现评分数据，不要改变内容和例子或创造新的数据，字数150字以内：
${salesData.理由}`;

        const response = await axios.post<EvaluationResponse>('http://localhost:3001/api/evaluate', {
            message
        }, {
            timeout: 30000 // 设置30秒超时
        });

        if (!response.data?.result) {
            console.error('Invalid API response format:', response.data);
            return '评价生成失败：API 返回格式异常';
        }

        return response.data.result || '暂无评价内容';
    } catch (error: any) {
        if (error?.isAxiosError) {
            if (error.code === 'ECONNABORTED') {
                console.error('API request timeout:', error);
                return '评价生成超时，请稍后重试';
            }
            if (error.response) {
                console.error('API error response:', error.response.data);
                return `评价生成失败：${error.response.status} - ${error.response.statusText}`;
            }
            if (error.request) {
                console.error('No API response:', error.request);
                return '评价生成失败：无法连接到服务器';
            }
        }
        console.error('Error getting AI evaluation:', error);
        return '评价生成失败，请稍后重试';
    }
}; 

File: src/utils/ai_evaluator.py
import warnings
warnings.filterwarnings("ignore")
from openai import OpenAI
from flask import Flask, request, jsonify
from flask_cors import CORS
import time

app = Flask(__name__)
CORS(app)

def chat(message):
    client = OpenAI(
        api_key="eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6IiIsInV1aWQiOiJ1SiVEMVhaSW91UFlwJCprLTlmNWMxYmNkLWZhOWYtNGM5OC1iNTAxLTlhYTcxZTZmNjMzOSJ9.4YOGpWUHRLPthGkKJsAVG31Xk3ZyEMOtsptLOIaRpT4",
        base_url="https://muses.weizhipin.com/muses-open/openai/v1",
        timeout=25.0  # 设置25秒超时
    )
    messages = message 
    try:
        start_time = time.time()
        stream = client.chat.completions.create(
            model="nanbeige2-turbo-0918",
            messages=[{"role": "user", "content": f'{messages}'}],
            max_tokens=4096,
            temperature=0.6,
            stream=False,
        )
        end_time = time.time()
        print(f"API响应时间: {end_time - start_time:.2f}秒")
        text = stream.choices[0].message.content
        print(f"原始AI响应: {text}")
        
        return text
    except Exception as e:
        print(f"API调用出错: {str(e)}")
        raise e

@app.route('/api/evaluate', methods=['POST'])
def evaluate():
    data = request.json
    if not data or 'message' not in data:
        return jsonify({'error': '缺少必要的参数'}), 400
    
    try:
        result = chat(data['message'])
        return jsonify({'result': result})
    except Exception as e:
        print(f"评估请求处理出错: {str(e)}")
        return jsonify({'error': str(e)}), 500

if __name__ == '__main__':
    app.run(port=3001) 